hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/RecoverableZooKeeper.java

```diff
@@ -205,12 +203,21 @@ public void delete(String path, int version) throws InterruptedException, Keeper
    * @return A Stat instance
    */
   public Stat exists(String path, Watcher watcher) throws KeeperException, InterruptedException {
+    return getStatInstance(path, watcher, null);
+  }
+
+  private Stat getStatInstance(String path, Watcher watcher, Boolean watch)
+          throws InterruptedException, KeeperException {
     try (TraceScope scope = TraceUtil.createTrace("RecoverableZookeeper.exists")) {
```

Comment:
Why is there a need for the `scope` variable? It isn't used anywhere.

Reply:
It’s true that the scope variable isn’t used anywhere. However, that is how it is used with try-with-resources in most of the cases in this project to create htrace and then close it.
Are you suggesting to keep it this way: try(TraceUtil.createTrace())  ?

Reply:
@HorizonNet